Navbar component created

Efekahya opened the pull request:
![image](https://user-images.githubusercontent.com/75262924/180977170-6592645c-3339-463d-92c6-179d25931d0e.png)
https://www.figma.com/file/SBPCNjmePIwJCzqqhXF9Y8/Rick-and-Morty---Digieggs-Assessment?node-id=1027%3A3541

Efekahya committed:
Navbar component created

bybash commented on src/index.tsx:
```diff
@@ -3,15 +3,25 @@ import ReactDOM from 'react-dom/client';
+
+const client = new ApolloClient({
+  uri: "https://rickandmortyapi.com/graphql",
+
```
unnecessary new line.

bybash commented on src/index.tsx:
```diff
@@ -3,15 +3,25 @@ import ReactDOM from 'react-dom/client';
 import "./styles.scss"
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import { ApolloClient, InMemoryCache, ApolloProvider } from "@apollo/client"
```
Please sort your imports. I wrote the order how it should be in your other PR, check it.

bybash commented on src/components/Navbar/styles.scss:
```diff
@@ -9,9 +9,10 @@
 .wrapper {
-    width: 70%;
+    width: 100%;
+        max-width: 1162px;
```
indent issue

bybash commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,83 @@
+          </div>
+        </div>
+      </nav>
+      <div className="secondaryInput">
```
Why do you have another search input here?

Efekahya replied:
I'm hiding the main search bar on breakpoint and displaying the second search bar.

iremlaya replied:
Rather than that, create a SearchBar component that is controlled by parent state to reduce code repeat

bybash replied:
I don't know mobile design if it is something that you can handle by CSS then don't write redundant codes. Nevertheless, this is doable if designs are entirely different and impossible to make by CSS.

bybash commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,83 @@
+        name: e.target.value,
+      },
+    })
+    if (data) {
+      console.log(data.characters.info.count, data.characters.results)
+    }
```
Don't forget to remove your console log codes before sending your PRs to review.

bybash commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,83 @@
+      console.log(data.characters.info.count, data.characters.results)
+    }
+  }
+  const QUERY = gql`
+    query Characters($filter: FilterCharacter!) {
+      characters(filter: $filter) {
+        info {
+          count
+        }
+        results {
+          name
+        }
+      }
+    }
+  `
+  type FilterCharacter = {
+    name: String
+  }
+  const filter: FilterCharacter = {
+    name: "rick",
+  }
```
Define these bits in other files

Efekahya committed:
Moved interfaces, created a search bar component

Formatted documents and sorted import order

Changes made to the app.tsx can be ignored!

bybash commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,69 @@
+      },
+    })
+  }
+  const QUERY = gql`
+    query Characters($filter: FilterCharacter!) {
+      characters(filter: $filter) {
+        info {
+          count
+        }
+        results {
+          name
+        }
+      }
+    }
+  `
```
Take this query into a separate file.

bybash commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,69 @@
   const [searchValue, setSearchValue] = React.useState("")
 
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value)
+    refetch({
+      filter: {
+        name: e.target.value,
+      },
+    })
```
It would be good to have a debounce search here. Otherwise it will search everytime you hit a button.

Efekahya committed:
fix style

Efekahya committed:
debounce search

atvdi commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,74 @@
+          <SearchBar
+            handleSearch={(e) => debounceFn(e.target.value)}
+            searchValue={searchValue}
+            classValue={"searchInput"}
```
```suggestion
            classValue="searchInput"
```

atvdi commented on src/components/Navbar/index.tsx:
```diff
@@ -1,40 +1,74 @@
+        <SearchBar
+          handleSearch={(e) => debounceFn(e.target.value)}
+          searchValue={searchValue}
+          classValue={"searchInputTwo"}
```
```suggestion
          classValue="searchInputTwo"
```

atvdi reviewed (changes requested):
Nice

Efekahya committed:
Update src/components/Navbar/index.tsx

Co-authored-by: Abdullah Tanrıverdi <89466000+loneddonkey@users.noreply.github.com>

Efekahya committed:
Update src/components/Navbar/index.tsx

Co-authored-by: Abdullah Tanrıverdi <89466000+loneddonkey@users.noreply.github.com>
